### PR TITLE
Add section to README discussing the variations for handling optional values

### PR DESCRIPTION
**Pull request by kestred**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -239,6 +239,49 @@ divide(10, 2); // 5
...
+```ts
+// For variables that might be undefined or null
+const MyString: String;                    // string             (e.g. 'text')
+const MyStringMaybe: String.Or(Undefined); // string | undefined (e.g. 'text', undefined)
+const MyStringNullable: String.Or(Null);   // string | null      (e.g. 'text', null)
+const MyStringVoidable: String.Or(Void);   // string | void      (e.g. 'text', null, undefined)
+```
+
+If a `Record` may or may not have some keys, we can declare the optional
+keys using `myRecord.Partial`.  Partial keys validate successfully if
```
Comment (pelotom) [inline README.md]: I think you meant

```suggestion
keys using `myRecord.And(Partial({ ... }))`.  Partial keys validate successfully if
```

File: README.md [hunk 2]
```diff
@@ -239,6 +239,49 @@ divide(10, 2); // 5
...
+they are absent or undefined (but not null) or the type specified
+(which can be null).
+
+```ts
+// Using `Ship` from above
+const RegisteredShip = Ship.And({
+  // All registered ships must have this flag
+  isRegistered: Literal(true),
+}).Partial({
```
Comment (pelotom) [inline README.md]:
```suggestion
}).And(Partial({
```

File: README.md [hunk 3]
```diff
@@ -239,6 +239,49 @@ divide(10, 2); // 5
...
+  // We may or may not know the ship's classification
+  name: Union(Literal('military'), Literal('civilian')),
+
+  // We may not know the ship's rank (undefined),
+  // we may also know that a civilian ship doesn't have a rank (null)
+  rank: Rank.Or(Null),
+});
```
Comment (pelotom) [inline README.md]:
```suggestion
}));
```

File: README.md [hunk 4]
```diff
@@ -239,6 +239,49 @@ divide(10, 2); // 5
...
+const RegisteredShip = Ship.And({
+  // All registered ships must have this flag
+  isRegistered: Literal(true),
+}).Partial({
+  // We may or may not know the ship's classification
+  name: Union(Literal('military'), Literal('civilian')),
+
+  // We may not know the ship's rank (undefined),
+  // we may also know that a civilian ship doesn't have a rank (null)
+  rank: Rank.Or(Null),
```
Comment (pelotom) [inline README.md]: The comment says "We may not know the ship's rank (undefined) ..." but this code will not allow for `undefined`.

**Comment by kestred** [comment]
Oops, yup!  A few mistakes here and there.  I should have time later in the week for another pass.

File: README.md (hunk 4 above)
Comment (pelotom) [reply README.md]: I see, yes that makes sense.

**Comment by pelotom** [comment]
Thanks!

**Pull request merged**